vhost-device-sound/src/device.rs

```diff
@@ -155,86 +155,58 @@ impl VhostUserSoundThread {
         let mut any = false;
 
         for desc_chain in requests {
-            let descriptors: Vec<_> = desc_chain.clone().collect();
-            if descriptors.len() < 2 {
-                return Err(Error::UnexpectedDescriptorCount(descriptors.len()).into());
-            }
+            let mem = atomic_mem.memory();
 
-            // Request descriptor.
-            let desc_request = descriptors[0];
-            if desc_request.is_write_only() {
-                return Err(Error::UnexpectedWriteOnlyDescriptor(0).into());
-            }
+            let mut reader = desc_chain
+                .clone()
+                .reader(&mem)
+                .map_err(|_| Error::DescriptorReadFailed)?;
```

Comment:
Maybe here we can reuse the `UnexpectedWriteOnlyDescriptor` error.

Reply:
I agree. However, the `UnexpectedWriteOnlyDescriptor` error requires as parameter the index of the descriptor that should not be `WriteOnly` but there is not such information. `reader` returns always `InvalidChain` in case or error. Should I keep `DescriptorReadFailed`?

Reply:
After this PR `UnexpectedWriteOnlyDescriptor` is not used in any place, so you can adapt it or remove it if you prefer to use `DescriptorReadFailed`. I don't have a strong opinion, so it's up to you ;-)

Reply:
is it OK if I add a new commit to remove the unused errors?

Reply:
Yep, sure.